packages/@react-aria/grid/src/useGridSelectionCheckbox.ts

```diff
@@ -31,8 +32,30 @@ export function useGridSelectionCheckbox<T, C extends GridCollection<T>>(props:
   let isDisabled = !state.selectionManager.canSelectItem(key);
   let isSelected = state.selectionManager.isSelected(key);
 
+  let isShiftDown = useRef(false);
+  useEffect(() => {
+    let trackKeyDown = (e: KeyboardEvent) => {
+      if (e.key === 'Shift') {
+        isShiftDown.current = true;
+      }
+    };
+    let trackKeyUp = (e: KeyboardEvent) => {
+      if (e.key === 'Shift') {
+        isShiftDown.current = false;
+      }
+    };
+
+    document.addEventListener('keydown', trackKeyDown);
+    document.addEventListener('keyup', trackKeyUp);
+
+    return () => {
+      document.removeEventListener('keydown', trackKeyDown);
+      document.removeEventListener('keyup', trackKeyUp);
+    };
+  });
```

Comment:
```suggestion
  }, []);
```
We don't want to re-run this every time we render, only on mount/unmount

Reply:
oh my... now idea how I missed that. added.